sycl/test-e2e/InvokeSimd/Feature/invoke_simd_struct_by_pointer.cpp

```diff
@@ -0,0 +1,159 @@
+// TODO: enable on Windows once driver is ready
+// REQUIRES: gpu && linux
+// UNSUPPORTED: cuda || hip
+//
+// Check that full compilation works:
+// RUN: %clangxx -fsycl -fno-sycl-device-code-split-esimd -Xclang -fsycl-allow-func-ptr %s -o %t.out
+// RUN: env IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %GPU_RUN_PLACEHOLDER %t.out
+//
+// VISALTO enable run
+// RUN: env IGC_VISALTO=63 IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %GPU_RUN_PLACEHOLDER %t.out
+
+// Tests invoke_simd support in the compiler/headers
+
+/* Test case specification:
+ * -----------------------
+ * Test passing pointers to structs to invoke_simd functions.
+ *
+ * Test case description:
+ * -----------------------------
+ * This is a simple test that defines a struct called 'multipliers' with 2 int
+ * members, x and y. The pointer to an instance of this structure is passed to
+ * a function called by invoke_simd. This function multiplies the vector B by
+ * both x and y and adds the resulting value to the vector A. The result of the
+ * addition is returned and stored in C.
+ */
+
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/ext/oneapi/experimental/invoke_simd.hpp>
+#include <sycl/sycl.hpp>
+
+#include <functional>
+#include <iostream>
+#include <type_traits>
+
+/* Subgroup size attribute is optional.
+ * In case it is absent compiler decides what subgroup size to use.
+ */
+#ifdef IMPL_SUBGROUP
+#define SUBGROUP_ATTR
+#else
+#define SUBGROUP_ATTR [[intel::reqd_sub_group_size(VL)]]
+#endif
+
+using namespace sycl::ext::oneapi::experimental;
+namespace esimd = sycl::ext::intel::esimd;
+constexpr int VL = 16;
+
+struct multipliers {
+  int x;
+  int y;
+};
+
+/* Performs: A + B * scalars->x * scalars->y. */
+__attribute__((always_inline)) esimd::simd<int, VL>
+ESIMD_CALLEE(int *A, esimd::simd<int, VL> b, int i,
+             multipliers *scalars) SYCL_ESIMD_FUNCTION {
+  esimd::simd<int, VL> a;
+  a.copy_from(A + i);
+  return a + b * scalars->x * scalars->y;
+}
+
+[[intel::device_indirectly_callable]] SYCL_EXTERNAL
+    simd<int, VL> __regcall SIMD_CALLEE(int *A, simd<int, VL> b, int i,
```

Comment:
Is it `clang-format` that results in this ugly formatting?

Reply:
Sadly yes.